### ISSUE TITLE
POC: create /etc/resolv.conf from the initrd

### ISSUE THREAD

**Issue by moul** [online-labs/initrd#33]

**Comment by moul** [comment]
And also /etc/hosts + /etc/hostname

**Comment by pisto** [comment]
I'm not sure you want to do that, just like the timezone I'd like to be able to set persistently the hostname.

**Comment by moul** [comment]
In a typical server I agree, but I think in a cloud server, you can imagine to use the server name as hostname, however, it's easy for me to update /etc/hostname only if it's empty.
As I can only `sed` hostnames and dns in /etc/hosts and /etc/resolv.conf instead of overwriting the complete files, to preserve user data

**Comment by moul** [comment]
What do you think about:

- if /etc/hostname exists in rootfs, the initrd do nothing, else, it fills the entire file with hostname from metadatas
- if /etc/resolv.conf exists and contains nameserver entries, it let the file as it, else, it appends the nameserver= values
- if /etc/hosts exists, it only replaces the current hostname from /etc/hostname with the actual 212.x.x.x ip address ?

**Comment by pisto** [comment]
* ok
* it's not clear to me what NetworkManager tries to do after jumping out of initrd, so I I have no idea what the correct approach is here
* why do you need to inject a hostname pointing to self? if it's just for "aesthetics", I'd just avoid that at all.

**Comment by aimxhaisse** [comment]
I'm 100% OK with this: moving this kind of work from each image to the initrd is a good way to stop hacking a lot of stuff in  distributions without going into the way of advanced users; that will limit the amount of tricks we have in each distributions (and introduce dependencies like: "you have to use networkmanager in Archlinux because "we" decided that we were going to use it to introduce our DNS).

:ok_hand:

**Comment by moul** [comment]
About NetworkManager, I tried to play a lot with options, but in case eth0 is already mounted, it does not trigger all the hooks

The needs to inject an ip with self hostname is for some softwares like apache that fails when starting to listen (a getaddr* function issue afaik)

---

This issue was revealed with failed attempt to bump fedora to 21, but the approach is more global and is already used by Docker on all kind of linux image: https://github.com/docker/docker/blob/478a9462e8776f60c8445972bcb3e506c6e7d413/daemon/volumes.go#L335-L345

**Comment by pisto** [comment]
I just wiped my /etc/hosts and put
```
127.0.0.1	localhost
::1		localhost
```
and httpd started just fine?

**Comment by pisto** [comment]
Maybe I understand what you mean:
```
AH00558: httpd: Could not reliably determine the server's fully qualified domain name, using fe80::207:cbff:fe03:6724. Set the 'ServerName' directive globally to suppress this message
```
However I think this is just normal and harmless, and putting a entry for the external ip (212.*) won't solve it.

**Comment by moul** [comment]
Ok, for now I prepare the code but I don't enable it for /etc/hosts

Are you okay for:
- smart /etc/hostname (not already existing or empty only)
- smart /etc/resolv.conf (not erasing, just appending nameservers)

?

**Comment by pisto** [comment]
yes

**Comment by moul** [comment]
Perfect :)

**Comment by moul** [comment]
Fixed by #156
